test/functional/tests/io/test_write_fetch.py

```diff
@@ -0,0 +1,210 @@
+                pattern=pattern)
+        if CacheModeTrait.InsertRead not in cache_mode_traits:
+            cache.flush_cache()
+        casadm.reset_counters(cache.cache_id, core.core_id)
+    with TestRun.step("Run writes to CAS device using fio."):
+        io_stats_before_io = cache_disk.get_io_stats()
+        run_fio(target=core.system_path,
+                operation_type=ReadWrite.write,
+                skip=cache_line_size.value / 2,
+                blocksize=cache_line_size.value / 2 * 3,
+                io_size=io_size)
+    with TestRun.step("Verify CAS statistics for partial misses, write hits and writes to cache."):
+        check_statistics(cache=cache,
+                         io_interval=cache_line_size.value * 2,
+                         io_size=io_size,
+                         partial_misses=True)
+    with TestRun.step("Verify number of writes to cache device using iostat. Shall be 450M "
+                      "(+ metadata for cache mode with write insert feature)."):
+        check_io_stats(cache_disk=cache_disk,
+                       cache=cache,
+                       io_stats_before=io_stats_before_io,
+                       io_size=io_size,
+                       partial_misses=True)
+
+
+# Methods used in tests:
+def check_io_stats(cache_disk, cache, io_stats_before, io_size, partial_misses=False):
+    io_size_mib = int(io_size.get_value(Unit.MebiByte))
+    io_stats_after = cache_disk.get_io_stats()
+    logical_block_size = int(TestRun.executor.run(
+        f"cat /sys/block/{cache_disk.device_name}/queue/logical_block_size").stdout)
+    diff = io_stats_after.sectors_written - io_stats_before.sectors_written
+    written_sector_size_mib = int((Size(logical_block_size) * diff).get_value(Unit.MebiByte))
+    TestRun.LOGGER.info(f"Sectors written: "
+                        f"{io_stats_after.sectors_written - io_stats_before.sectors_written} "
+                        f"({written_sector_size_mib}MiB)")
+
+    expected_writes_mib = io_size_mib / 2
+    if partial_misses:
+        expected_writes_mib += io_size_mib / 4
+
+    cache_line_size = int(cache.get_cache_line_size().value.get_value(Unit.KibiByte))
+    cache_mode_traits = CacheMode.get_traits(cache.get_cache_mode())
+
+    if CacheModeTrait.InsertWrite | CacheModeTrait.LazyFlush in cache_mode_traits:
+        # Metadata size is 4KiB per each cache line
+        metadata_size = io_size_mib * 4 / cache_line_size
+        expected_writes_mib += metadata_size
+
+    if not validate_value(int(expected_writes_mib), written_sector_size_mib):
+        TestRun.LOGGER.error(f"IO stat writes to cache ({written_sector_size_mib}MiB) "
+                             f"inconsistent with expected value ({expected_writes_mib}MiB)")
+
+
+def validate_value(expected, actual):
+    if expected == 0:
+        return actual == 0
+    val = abs(100 * actual / expected - 100)
+    return val < 1
+
+
+def check_statistics(cache, io_interval, io_size, partial_misses=False):
```

Comment:
Instead of `io_interval` I would pass block_size and skip_size. Expected number of requests would be equal to block_size, and `expected_writes` would be equal to (block_size)/(block_size+skip_size) so `if` in line 187 could be removed.